src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -40,21 +51,20 @@ impl ScalarFunc {
         ctx: &mut EvalContext,
         row: &'a [Datum],
     ) -> Result<Option<Cow<'a, Time>>> {
-        let mut t = match self.children[0].eval_time(ctx, row) {
-            Err(err) => return Error::handle_invalid_time_error(ctx, err),
-            Ok(None) => {
-                return Error::handle_invalid_time_error(
-                    ctx,
-                    Error::incorrect_datetime_value("None"),
-                )
-            }
-            Ok(Some(res)) => res,
-        };
+        let mut t = direct_try!(
```

Comment:
Sorry, but I don't see the reason of introducing "direct_try!" here. Could you further explain what you wanted to do to me?

I guess you can write:

```rust
let mut t = self.children[0]
    .eval_time(ctx, row)
    .and_then(|some_time| some_time.ok_or(Error::incorrect_datetime_value("None")))   // turn None into Error and Some into Ok
    .or_else(|e| Error::handle_invalid_time_error(ctx, e).and_then(|_| Ok(None)))?;  // handle invalid time if there are errors. If errors are turned into warnings, use None as the value
```

If you meet syntax errors when trying to do so, please let me know (because I didn't feed it to a compiler).